File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/25.go
package main
import (
    "fmt"
    "strings"
)

func aAntesDeB(s string) bool {
    return !strings.Contains(s, "ba")
}

func main() {
    fmt.Println(aAntesDeB("aaabbb"))  // True
    fmt.Println(aAntesDeB("abba"))    // False
}

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/25.java
public class AFNAAntesDeB {
    public static boolean aAntesDeB(String s) {
        return !s.contains("ba");
    }

    public static void main(String[] args) {
        System.out.println(aAntesDeB("aaabbb"));  // True
        System.out.println(aAntesDeB("abba"));    // False
    }
}
